contents/blog/cypress-end-to-end-tests.md

```diff
@@ -0,0 +1,233 @@
+---
+date: 2020-06-10
+title: Merge with Confidence
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+---
+
+## Moving fast is easy. Moving fast with confidence is hard.
+
+If you've been keeping track of the array blog posts you know that we prioritize shipping things fast and often. Just as important to us is being sure that we are not going to break things unneccesarily for our users as we add new features and speedups. What we have found that works really well is nothing terribly novel by itself: a solid foundation of unit tests, end to end tests (integration tests), and CI/CD that for automation and gatekeeper keeping master clean.
+
+### Unit & Integration tests
+
+In our Django codebase you'll find good bit of [Django tests](https://github.com/PostHog/posthog/tree/master/posthog/test) that help keep us honest as we hack away at the backend of Posthog that keeps track of all the 1's and 0's that our customers depend on for making product decisions. These are our frontline defenders that let us know that something might be up before we even get to the point of creating a PR.
+
+For this we do lean heavily on the standard Django test runner.
+
+#### Running Django tests locally
+
+```bash
+python manage.py test
```

Comment:
I'm not sure about the value of having this line here. It makes me think this is documentation, and not a blog post.

Reply:
Agreed. I removed it with the commit below.